aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3427,15 +3429,36 @@ static void linalg_lstsq_out_info(
   auto input_working_copy = copyBatchedColumnMajor(input);
 
   // now the actual call that computes the result in-place (apply_lstsq)
-  lstsq_stub(input.device().type(), input_working_copy, solution, rank, singular_values, infos, rcond, driver);
-
+  if (driver == "gelss" && input.device() != at::kCPU) {
+    if (input.numel() == 0) {
+        auto output_shape = input.sizes().vec();
+        output_shape.back() = other.size(-1);
+        solution.zero_();
+    } else {
+        auto [U, S, Vh] = at::_linalg_svd(input, false, true, "gesvd");
+        rank = at::zeros({1}, at::kLong);
+        rank[0] = (S > rcond).sum();
```

Comment:
This is not correct. Compute the rank by looking at the zeros of `S_pinv`.

Reply:
For my understanding(hopefully not a stupid question) - is it because the way it's done currently is unstable or it leads to inaccurate results?

Reply:
It's not about being unstable. It's about the way we define what is a zero singular value in this function.